File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/virology/exchange/CtaExchangeResponseV2.java
package uk.nhs.nhsx.virology.exchange;

import uk.nhs.nhsx.virology.TestKit;

public class CtaExchangeResponseV2 {

    public final String diagnosisKeySubmissionToken;
    public final String testResult;
    public final String testEndDate;
    public final TestKit testKit;
    public final Boolean diagnosisKeySubmissionSupported;
    public final Boolean requiresConfirmatoryTest;

    public CtaExchangeResponseV2(String diagnosisKeySubmissionToken,
                                 String testResult,
                                 String testEndDate,
                                 TestKit testKit,
                                 Boolean diagnosisKeySubmissionSupported,
                                 Boolean requiresConfirmatoryTest) {
        this.diagnosisKeySubmissionToken = diagnosisKeySubmissionToken;
        this.testResult = testResult;
        this.testEndDate = testEndDate;
        this.testKit = testKit;
        this.diagnosisKeySubmissionSupported = diagnosisKeySubmissionSupported;
        this.requiresConfirmatoryTest = requiresConfirmatoryTest;
    }

}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/virology/exchange/CtaExchangeResponseV2.kt
package uk.nhs.nhsx.virology.exchange

import uk.nhs.nhsx.domain.DiagnosisKeySubmissionToken
import uk.nhs.nhsx.domain.TestEndDate
import uk.nhs.nhsx.domain.TestKit
import uk.nhs.nhsx.domain.TestResult

data class CtaExchangeResponseV2(
    val diagnosisKeySubmissionToken: DiagnosisKeySubmissionToken,
    val testResult: TestResult,
    val testEndDate: TestEndDate,
    val testKit: TestKit,
    val diagnosisKeySubmissionSupported: Boolean,
    val requiresConfirmatoryTest: Boolean,
    val confirmatoryDayLimit: Int?,
    val venueHistorySharingSupported: Boolean = false
)
